src/library/scala/Function.scala

```diff
@@ -27,6 +27,14 @@ object Function {
   /** The constant function */
   def const[T, U](x: T)(y: U): T = x
 
+  def constLazy[T, U](x: => T)(y: U): T = x
```

Comment:
docs would be nice, particularly for this one which one might think should only evaluate `x` once (I realise that's what `constMemoized` does). I think this is probably the best name there is for it though.

Reply:
Yeah, the naming convention I've seen is "lazy" for "by-name".  It doesn't help that the `lazy` keyword is only applicable to `val`, to make it "by-need" or memoized laziness.  I'll doc it up.